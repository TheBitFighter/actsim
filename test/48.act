import globals;
import std;
open std::channel;

defproc src (a1of2! x)
{
  prs {
    Reset | x.a => x.t-
    Reset -> x.f-
  }
}

defproc sink (a1of2? x)
{
  int v;
  int i;
  chp {
    i := 0;
   *[ i < 100 -> x?v; log ("got ", v); i := i + 1]
  }
}

defproc test ()
{
   src s;
   sink t(s.x);
}

Initialize {
  actions { Reset+ };
  actions { Reset- }
}
