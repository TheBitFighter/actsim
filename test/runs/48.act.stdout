[                 130] <t>  got 1
[                 160] <t>  got 1
[                 190] <t>  got 1
[                 220] <t>  got 1
[                 250] <t>  got 1
[                 280] <t>  got 1
[                 310] <t>  got 1
[                 340] <t>  got 1
[                 370] <t>  got 1
[                 400] <t>  got 1
[                 430] <t>  got 1
[                 460] <t>  got 1
[                 490] <t>  got 1
[                 520] <t>  got 1
[                 550] <t>  got 1
[                 580] <t>  got 1
[                 610] <t>  got 1
[                 640] <t>  got 1
[                 670] <t>  got 1
[                 700] <t>  got 1
[                 730] <t>  got 1
[                 760] <t>  got 1
[                 790] <t>  got 1
[                 820] <t>  got 1
[                 850] <t>  got 1
[                 880] <t>  got 1
[                 910] <t>  got 1
[                 940] <t>  got 1
[                 970] <t>  got 1
[                1000] <t>  got 1
[                1030] <t>  got 1
[                1060] <t>  got 1
[                1090] <t>  got 1
[                1120] <t>  got 1
[                1150] <t>  got 1
[                1180] <t>  got 1
[                1210] <t>  got 1
[                1240] <t>  got 1
[                1270] <t>  got 1
[                1300] <t>  got 1
[                1330] <t>  got 1
[                1360] <t>  got 1
[                1390] <t>  got 1
[                1420] <t>  got 1
[                1450] <t>  got 1
[                1480] <t>  got 1
[                1510] <t>  got 1
[                1540] <t>  got 1
[                1570] <t>  got 1
[                1600] <t>  got 1
[                1630] <t>  got 1
[                1660] <t>  got 1
[                1690] <t>  got 1
[                1720] <t>  got 1
[                1750] <t>  got 1
[                1780] <t>  got 1
[                1810] <t>  got 1
[                1840] <t>  got 1
[                1870] <t>  got 1
[                1900] <t>  got 1
[                1930] <t>  got 1
[                1960] <t>  got 1
[                1990] <t>  got 1
[                2020] <t>  got 1
[                2050] <t>  got 1
[                2080] <t>  got 1
[                2110] <t>  got 1
[                2140] <t>  got 1
[                2170] <t>  got 1
[                2200] <t>  got 1
[                2230] <t>  got 1
[                2260] <t>  got 1
[                2290] <t>  got 1
[                2320] <t>  got 1
[                2350] <t>  got 1
[                2380] <t>  got 1
[                2410] <t>  got 1
[                2440] <t>  got 1
[                2470] <t>  got 1
[                2500] <t>  got 1
[                2530] <t>  got 1
[                2560] <t>  got 1
[                2590] <t>  got 1
[                2620] <t>  got 1
[                2650] <t>  got 1
[                2680] <t>  got 1
[                2710] <t>  got 1
[                2740] <t>  got 1
[                2770] <t>  got 1
[                2800] <t>  got 1
[                2830] <t>  got 1
[                2860] <t>  got 1
[                2890] <t>  got 1
[                2920] <t>  got 1
[                2950] <t>  got 1
[                2980] <t>  got 1
[                3010] <t>  got 1
[                3040] <t>  got 1
[                3070] <t>  got 1
[                3100] <t>  got 1
